File: Data-Structure-Codes-master/Quicksort.java
package com.company;

import java.util.*;


class Main{
    static Scanner input = new Scanner(System.in);
    static ArrayList<Integer> arrayList = new ArrayList<>();

    static void Quicksort(ArrayList<Integer> arrayList,int left,int right){
        int i=left;
        int j = right;
        int temp;
        int pivot = arrayList.get((left+right)/2);

        while (i<=j){
            while (arrayList.get(i)<pivot){
                i++;
            }
            while (arrayList.get(j)>pivot){
                j--;
            }
            if(i<=j){
                temp = arrayList.get(i);
                arrayList.set(i,arrayList.get(j));
                arrayList.set(j,temp);
                i++;
                j--;
            }
        }
        if(left<j)
            Quicksort(arrayList,left,j);
        if(i<right)
            Quicksort(arrayList,i,right);
    }





    public static void main(String[] args) {

        System.out.println("Enter the number of element:");
        int numOfelement = input.nextInt();
        System.out.printf("Enter the values:\n");

        for(int i=0;i<numOfelement;i++){
            int value = input.nextInt();
            arrayList.add(value);
        }

        Quicksort(arrayList,0,arrayList.size()-1);

        System.out.println("The sorted elements are:");
        for(int i=0;i<numOfelement;i++){
            System.out.printf("%d ",arrayList.get(i));
        }
    }
}

File: Data-Structure-Codes-master/quicksort.php
<?php

$array_to_sort = [1,4,2,9,6,2,7];

function quickSort(& $array,$left,$right){
    $i = $left;
    $j = $right;
    $pivot = $array[intval(($left+$right)/2)];

    while ($i<=$j){
        while ($array[$i]<$pivot){
            $i++;
        }

        while ($array[$j]>$pivot){
            $j--;
        }

        if ($i<=$j){
            $temp = $array[$i];
            $array[$i]=$array[$j];
            $array[$j] = $temp;
            $i++;
            $j--;
        }
    }

    if($left<$j){
        quickSort($array,$left,$j);
    }

    if($i<$right){
        quickSort($array,$i,$right);
    }
}

quickSort($array_to_sort,0,sizeof($array_to_sort)-1);

print_r($array_to_sort);

?>